Microsoft.Research/Contracts/PresentationFramework/Sources/System.Windows.Controls.Panel.cs

```diff
@@ -43,10 +43,11 @@ abstract public partial class Panel : System.Windows.FrameworkElement, System.Wi
     #region Methods and constructors
     protected virtual new UIElementCollection CreateUIElementCollection(System.Windows.FrameworkElement logicalParent)
     {
-      return default(UIElementCollection);
+        Contract.Ensures(Contract.Result<UIElementCollection>() != null);
+        return default(UIElementCollection);
     }
 
-    protected override System.Windows.Media.Visual GetVisualChild(int index)
+      protected override System.Windows.Media.Visual GetVisualChild(int index)

```

Comment:
Sorry, it's hard to keep the old formatting with 2 spaces indent where the default in VS is 4 spaces. Are there any coding guidelines for CC project?